cmd/cluster-init/update_bootstrap_secrets.go

```diff
@@ -0,0 +1,342 @@
+		return err
+	}
+	sc.From[key] = value
+	return nil
+}
+
+func updateExistingRegistryPullCredentialsAllSecrets(c *secretbootstrap.Config, o options) error {
+	for _, cluster := range c.UserSecretsTargetClusters {
+		if cluster != string(api.ClusterHive) && cluster != string(api.ClusterARM01) && cluster != o.clusterName {
+			return appendRegistrySecretItemContext(c, regPullCredsAll, cluster, secretbootstrap.DockerConfigJSONData{
+				AuthField:   fmt.Sprintf("token_image-puller_%s_reg_auth_value.txt", o.clusterName),
+				Item:        buildUFarm,
+				RegistryURL: registryUrlFor(o.clusterName),
+			})
+		}
+	}
+	return nil
+}
+
+func generateRegistryPullCredentialsAllSecrets(c *secretbootstrap.Config, o options) {
+	items := []secretbootstrap.DockerConfigJSONData{
+		{
+			AuthField:   "token_image-puller_ci_reg_auth_value.txt",
+			Item:        buildUFarm,
+			RegistryURL: api.ServiceAPPCIRegistry,
+		},
+		{
+			AuthField:   "auth",
+			Item:        "cloud.openshift.com-pull-secret",
+			RegistryURL: "cloud.openshift.com",
+			EmailField:  "email",
+		},
+		{
+			AuthField:   "auth",
+			Item:        "quay.io-pull-secret",
+			RegistryURL: "quay.io",
+			EmailField:  "email",
+		},
+		{
+			AuthField:   "auth",
+			Item:        "registry.connect.redhat.com-pull-secret",
+			RegistryURL: "registry.connect.redhat.com",
+			EmailField:  "email",
+		},
+		{
+			AuthField:   "auth",
+			Item:        "registry.redhat.io-pull-secret",
+			RegistryURL: "registry.redhat.io",
+			EmailField:  "email",
+		},
+	}
+	for _, cluster := range c.UserSecretsTargetClusters {
+		if cluster != string(api.ClusterHive) {
+			items = append(items, secretbootstrap.DockerConfigJSONData{
+				AuthField:   registryPullTokenField(cluster),
+				Item:        buildUFarm,
+				RegistryURL: registryUrlFor(cluster),
+			})
+		}
+	}
+	sc := secretbootstrap.SecretConfig{
+		From: map[string]secretbootstrap.ItemContext{
+			dotDockerConfigJson: generatePushPullSecretFrom(o.clusterName, items),
+		},
+		To: []secretbootstrap.SecretContext{
+			generateDockerConfigJsonSecretConfigTo(regPullCredsAll, ci, o.clusterName),
+			generateDockerConfigJsonSecretConfigTo(regPullCredsAll, testCredentials, o.clusterName),
+		},
+	}
+	logrus.Infof("Creating new secret with 'to' of: %v", sc.To)
+	c.Secrets = append(c.Secrets, sc)
+}
+
+func registryUrlFor(cluster string) string {
+	switch cluster {
+	case string(api.ClusterVSphere):
+		return "registry.apps.build01-us-west-2.vmc.ci.openshift.org"
+	case string(api.ClusterAPPCI):
+		return api.ServiceDomainAPPCIRegistry
+	case string(api.ClusterARM01):
+		return "registry.arm-build01.arm-build.devcluster.openshift.com"
+	default:
+		return fmt.Sprintf("registry.%s.ci.openshift.org", cluster)
+	}
+}
+
+func appendRegistrySecretItemContext(c *secretbootstrap.Config, name string, cluster string, value secretbootstrap.DockerConfigJSONData) error {
+	logrus.Infof("Appending registry secret item to: {name: %s, cluster: %s}", name, cluster)
```

Comment:
When presenting structured data in logging, use fields:

```go
logrus.WithFields(logrus.Fields{
    "name": name,
    "cluster": cluster,
}).Info("Appending registry secret item.")
```

This then lets us filter and process the logs quickly and easily

Reply:
Ah thats cool to know!